### PR TITLE
Refactor problem_utils and Problem creation

### PR DESCRIPTION
**Pull request by T-Davis**
There isn't an issue associated with this one.
Have a look, let me know what you think

### CODE REVIEW

**Review by tamari-gray — APPROVED** [review]
woooh lgtm!

**Review by nickmeinhold — CHANGES_REQUESTED** [review]
Looks awesome mate!  I would happily approve but just had a couple of q's

File: lib/utils/problem_utils.dart [hunk 1]
```diff
@@ -25,171 +25,146 @@ import 'package:crowdleague/models/problems/update_leaguer_problem.dart';
...
     map = {
-  AppleSignInProblem: (dynamic error, trace, info) => AppleSignInProblem.by(
-        (b) => b
-          ..message = error.toString()
-          ..trace = trace.toString()
-          ..info = info,
+  AppleSignInProblem: (dynamic error, trace, info) => AppleSignInProblem(
+        message: error.toString(),
+        trace: trace.toString(),
+        info: info == null ? null : BuiltMap(info),
```
[the remark this answers is not in the document]
Comment (T-Davis) [reply lib/utils/problem_utils.dart:33]: You got it!
It took me a little bit to track down what was happening, but a runtime error it was. BuiltMap(null) is no good.

File: lib/utils/problem_utils.dart [hunk 2]
```diff
@@ -25,171 +25,146 @@ import 'package:crowdleague/models/problems/update_leaguer_problem.dart';
...
       ),
   ConnectToMessagesProblem: (dynamic error, trace, info) =>
-      ConnectToMessagesProblem.by(
-        (b) => b
-          ..message = error.toString()
-          ..trace = trace.toString()
-          ..info = info,
+      ConnectToMessagesProblem(
+        message: error.toString(),
```
[the remark this answers is not in the document]
Comment (T-Davis) [reply lib/utils/problem_utils.dart]: So good!
Fixed!

**Review by nickmeinhold — APPROVED** [review]
Beautiful work mate!!!

**Pull request merged**